src/slo/backup_quality/quality_sli_service.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from src.commons.big_query.big_query import BigQuery
+from src.slo.backup_quality.quality_query_specification import \
+  QualityQuerySpecification
+from src.slo.backup_quality.sli_table_newer_modification_predicate import \
+  SLITableNewerModificationPredicate
+from src.slo.sli_results_streamer import SLIResultsStreamer
+from src.slo.sli_view_querier import SLIViewQuerier
+
+
+class QualitySliService(object):
+
+    def __init__(self):
+        big_query = BigQuery()
+        self.querier = SLIViewQuerier(
+            big_query,  QualityQuerySpecification()
+        )
+        self.streamer = SLIResultsStreamer(
+            table_id="SLI_backup_quality"
+        )
+        self.table_newer_modification_predicate = SLITableNewerModificationPredicate(big_query)
+
+    def recalculate_sli(self):
+        logging.info("Recalculating quality SLI has been started.")
+
+        all_tables, snapshot_time = self.querier.query()
+        filtered_tables = [table for table in all_tables
+                           if self.__should_stay_as_sli_violation(table)]
+
+        logging.info("Quality SLI tables filtered from %s to %s", len(all_tables), len(filtered_tables))
+        self.streamer.stream(
+            filtered_tables,
+            snapshot_marker=self.__create_snapshot_marker_row(snapshot_time)
+        )
+
+    def __should_stay_as_sli_violation(self, table):
+        try:
+            return not self.table_newer_modification_predicate.is_modified_till_last_census_snapshot()
```

Comment:
I think it should be 'is_modified_since_last_census_snapshot'

Reply:
done